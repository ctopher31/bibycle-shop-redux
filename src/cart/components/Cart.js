import React from 'react';
import CartItems from './CartItems';
import CartSummary from './CartSummary';
import CartLineItem from './CartLineItem';

const Cart = ({ cart, cartCount, removeItem, shipping, subtotal, total }) => (
  <main className="content cart">
    <h1 className="page--heading">
      {cartCount > 0
        ? `Your Cart has ${cartCount} Item${cartCount > 1 ? 's' : ''}`
        : 'Your Cart is Empty'}
    </h1>
    <section className="cart--section">
      <CartItems>
        {cart.map(item => (
          <CartLineItem key={item.number} {...{ ...item, removeItem }} />
        ))}
      </CartItems>
      <CartSummary {...{ shipping, subtotal, total }} />
    </section>
  </main>
);

export default Cart;
