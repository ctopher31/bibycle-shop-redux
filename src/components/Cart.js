import React from 'react';
import { connect } from 'react-redux';
import CartItems from './CartItems';
import CartSummary from './CartSummary';
import LineItem from './LineItem';

const Cart = ({ cart, cartCount }) => (
  <main className="content cart">
    <h1 className="page--heading">
      {cartCount > 0
        ? `Your Cart has ${cartCount} Item${cartCount > 1 ? 's' : ''}`
        : 'Your Cart is Empty'
      }
    </h1>
    <section className="cart--section">
      <CartItems>
        {cart.map((item) => <LineItem key={item.number} {...item} />)}
      </CartItems>
      <CartSummary />
    </section>
  </main>
);

const mapStateToProps = state => ({
  cart: state.cart,
  cartCount: state.cartCount,
});

export default connect(mapStateToProps, null)(Cart);
