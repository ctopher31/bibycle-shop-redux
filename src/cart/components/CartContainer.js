import { connect } from 'react-redux';
import { removeItemAction } from '../sagas';
import Cart from './Cart';

const mapStateToProps = state => ({
  cart: state.cart,
  cartCount: state.cartCount,
  shipping: state.shipping,
  subtotal: state.subtotal,
  total: state.total,
});

const mapDispatchToProps = dispatch => ({ removeItem: key => dispatch(removeItemAction(key)) });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Cart);
