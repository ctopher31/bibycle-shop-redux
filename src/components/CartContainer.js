import React from 'react';
import { connect } from 'react-redux';
import { removeItem as removeItemAction } from '../actions/actions';
import Cart from './Cart';
import CartItems from './CartItems';
import CartSummary from './CartSummary';
import LineItem from './LineItem';

const CartContainer = ({
  cart,
  cartCount,
  removeItem,
  shipping,
  subtotal,
  total,
}) => (
  <Cart {...{ cartCount }}>
    <CartItems>
      {cart.map((item) => <LineItem key={item.number} {...{ ...item, removeItem }} />)}
    </CartItems>
    <CartSummary {...{ shipping, subtotal, total }} />
  </Cart>
);

const mapStateToProps = state => ({
  cart: state.cart,
  cartCount: state.cartCount,
  shipping: state.shipping,
  subtotal: state.subtotal,
  total: state.total,
});

const mapDispatchToProps = dispatch => ({ removeItem: number => dispatch(removeItemAction(number)), });

export default connect(mapStateToProps, mapDispatchToProps)(CartContainer);
