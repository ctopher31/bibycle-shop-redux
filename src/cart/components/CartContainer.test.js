import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import CartContainer from './CartContainer';
import Cart from './Cart';
import cartReducer from '../reducers';

describe('CartContainer', () => {
  it('should render the container component', () => {
    const subtotal = 32.95;
    const shipping = 15;
    const initialState = {
      cart: {
        items: [
          {
            number: '51001',
            name: 'widget',
            price: 35.57,
            salePrice: 32.95,
            onSale: true,
          },
        ],
        cartCount: 1,
        subtotal,
        total: subtotal > 0 ? subtotal + shipping : 0,
        shipping: subtotal > 0 ? shipping : 0,
      },
    };

    const store = createStore(cartReducer, initialState);

    const wrapper = mount(
      <Provider store={store}>
        <CartContainer />
      </Provider>
    );

    const container = wrapper.find(CartContainer);

    expect(container.length).toEqual(1);
    expect(container.find(Cart).length).toEqual(1);
    expect(store.getState().content).toMatchSnapshot();
  });
});
