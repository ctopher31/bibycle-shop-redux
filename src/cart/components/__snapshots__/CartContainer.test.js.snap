// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartContainer should render the container component 1`] = `undefined`;
